### ISSUE TITLE
Dialog "Copilot wants to sign-in" should not show

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#233424]
1. Have GH Copilot installed.
2. Sign out of all GH accounts in VS Code
3. Notice the Chat Welcome View asking you to Sign In to GH. Click it
4. There is a dialog if I will allow GitHub Copilot Chat to sign in using GitHub

![Image](https://github.com/user-attachments/assets/b8dcfdb8-8bde-48f7-9822-f8982289649a)

This dialog should not show because the user clicks on an action to Sign In.

fyi @bpasero

**Comment by TylerLeonhardt** [comment]
This modal ensures that the correct window is focused and the user is aware that they are going through an auth flow. Additionally, the next step of this opens the browser, and this modal prevents the user from being take to the browser out of band.

Picture an extension running a long task and the user does something else why they wait and at the end of that task it asks for auth... if they were doing something, to be thrown into their browser, they may have forgot why that happened. This makes it clear why it's happening.

Unless we had some way to ensure:
* The focused window triggered this
* The user interacted with _some_ UI component moments before

I don't think we should hide this modal.

This is similar to Safari's handling of opening a url in a new tab with browser APIs... if it isn't positive that you triggered opening a new tab moments ago, it will block the request to open a new tab so that the user isn't confused.

**Comment by isidorn** [comment]
Good points, thanks. Regarding

> Unless we had some way to ensure:
> 
> * The focused window triggered this
> * The user interacted with some UI component moments before

Could we do similar to Safari - so we detect if the user interacted on the focused window and in that case not show? I assume we would have to enrich the API for this (to pass some time of interaction, and focused window id)?

**Comment by TylerLeonhardt** [comment]
Yeah and on a related note @jrieken had opened https://github.com/microsoft/vscode/issues/158911#issuecomment-1355879846 a while back... which also talks about limiting when this is shown (although there, the intent is to not even do the auth flow, just showing something less in-your-face than the modal).

Nevertheless, we probably need some mechanism in the API stack to understand what the user was doing when `getSession` was called and I don't think there is any such thing at the moment... and I don't think letting the extension author pass something in is the right direction because they're likely to just abuse it/get it wrong... so it has to be transparent.
